data-pipeline/src/trace_exporter/mod.rs

```diff
@@ -588,38 +547,33 @@ impl TraceExporter {
     /// Add all spans from the given iterator into the stats concentrator
     /// # Panic
     /// Will panic if another thread panicked will holding the lock on `stats_concentrator`
-    fn add_spans_to_stats<'a>(&self, spans: impl Iterator<Item = &'a SpanBytes>) {
+    fn add_spans_to_stats(&self, collection: &TraceCollection) {
         if let StatsComputationStatus::Enabled {
             stats_concentrator,
             cancellation_token: _,
             exporter_handle: _,
         } = &**self.client_side_stats.load()
         {
             let mut stats_concentrator = stats_concentrator.lock().unwrap();
-            for span in spans {
-                stats_concentrator.add_span(span);
+            match collection {
+                TraceCollection::TraceChunk(traces) => {
+                    let spans = traces.iter().flat_map(|trace| trace.iter());
+                    for span in spans {
+                        stats_concentrator.add_span(span);
+                    }
+                }
+                // TODO: APMSP-18190
+                TraceCollection::V07(_) => todo!(),
             }
         }
     }
 
-    fn send_deser_ser(&self, data: tinybytes::Bytes) -> Result<String, TraceExporterError> {
-        // TODO base on input format
-        let (mut traces, _) = match msgpack_decoder::v04::from_slice(data) {
-            Ok(res) => res,
-            Err(err) => {
-                error!("Error deserializing trace from request body: {err}");
-                self.emit_metric(
-                    HealthMetric::Count(health_metrics::STAT_DESER_TRACES_ERRORS, 1),
-                    None,
-                );
-                return Err(TraceExporterError::Deserialization(err));
-            }
-        };
-
-        let num_traces = traces.len();
-
+    fn send_deser_ser(
```

Comment:
We can leave it as is in this PR but this function name is even less accurate after these changes.